turbopack/crates/turbo-rcstr/src/dynamic.rs

```diff
@@ -36,15 +42,23 @@ pub(crate) fn new_atom<T: AsRef<str> + Into<String>>(text: T) -> RcStr {
         return RcStr { unsafe_data };
     }
 
-    let entry = Arc::new(text.into());
+    let hash = compute_fxhash(text.as_ref());
+
+    let entry: Arc<TStr> = Arc::new(TStr(text.into(), hash));
```

Comment:
In general this could avoid the double indirection by using ThinArc:

https://docs.rs/triomphe/latest/triomphe/struct.ThinArc.html

But it should probably be a separate PR before this hash change to measure the difference.